src/pages/ContactPage/index.tsx

```diff
@@ -0,0 +1,15 @@
+import Contact from "../../components/ContactForm";
+import Footer from "../../components/Footer";
+import Header from "../../components/Header";
+
+function ContactPage() {
+    return (
+        <>
+            <Header onMainPage={false} />
```

Comment:
Good idea with using the `onMainPage` prop to display different content in Header. 

You could also use react-router-dom hook: https://stackoverflow.com/questions/66265608/react-router-v6-get-path-pattern-for-current-route inside components which needs to decide what to show. However your idea might be  better in this case :)